### PR TITLE
chore: Remove JS docs and miscellaneous

### PR DESCRIPTION
**Pull request by gaofanmichael**
- 1. remove unused JSDoc.
- 2. Change some miscellaneous regarding type, comments, importing style, etc.

### CODE REVIEW

File: samples/.eslintrc.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
 rules:
   no-console: off
+  node/no-unpublished-require: off
```
Comment (jmdobry) [inline samples/.eslintrc.yml]: Note, this file is auto-generated. Then next time `npm run generate-scaffolding` is run, these changes would be overwritten.

Your options are:

1) Add `--except eslintrc_samples` to the `generate-scaffolding` script, or
2) Talk to @lukesneeringer about getting your changes incorporated into the [eslintrc_samples template in nodejs-repo-tools](https://github.com/GoogleCloudPlatform/nodejs-repo-tools/blob/master/templates/nodejs/eslintrc_samples.tpl)

File: samples/quickstart.js [hunk 2]
```diff
@@ -21,10 +21,10 @@ const Logger = winston.Logger;
 const Console = winston.transports.Console;
 
 // Imports the Google Cloud client library for Winston
-const LoggingWinston = require('@google-cloud/logging-winston');
+const LoggingWinston = require('@google-cloud/logging-winston').LoggingWinston;
```
Comment (jmdobry) [inline samples/quickstart.js]: The following needs to be updated as well:

* https://github.com/googleapis/nodejs-logging-winston/blob/typescript/samples/setup_explicit.js

File: package.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
   "scripts": {
     "docs": "repo-tools exec -- jsdoc -c .jsdoc.js",
     "generate-scaffolding": "repo-tools generate all --except lib_readme",
-    "lint": "cd samples/ && npm link ../ && repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/",
+    "lint": "cd samples/ && npm install && npm link ../ && cd .. && repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/",
```
Comment (jmdobry) [inline package.json]: I think you should just change this to:

```
"lint": "repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/",
```

which is what all the other repos use.

File: samples/quickstart.js (hunk 2 above)
Comment (gaofanmichael) [reply samples/quickstart.js]: Done.

File: samples/.eslintrc.yml (hunk 1 above)
Comment (gaofanmichael) [reply samples/.eslintrc.yml]: removed.

**Pull request merged**